buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt

```diff
@@ -0,0 +1,436 @@
+        var output = ProcessOutput("", "", 0)
+        for (i in 1..times) {
+            val args = if (::multiArguments.isInitialized) arguments + multiArguments[i - 1] else arguments
+            output += if (::testData.isInitialized)
+                runProcessWithInput(project.executor::execute, executable, args, testData)
+            else
+                runProcess(project.executor::execute, executable, args)
+        }
+        if (compilerMessages) {
+            // TODO: as for now it captures output only in the driver task.
+            // It should capture output from the build task using Gradle's LoggerManager and LoggerOutput
+            val compilationLog = project.file("$executable.compilation.log").readText()
+            output.stdOut = compilationLog + output.stdOut
+        }
+        output.check()
+        output.print()
+    }
+
+    private operator fun ProcessOutput.plus(other: ProcessOutput) = ProcessOutput(
+            stdOut + other.stdOut,
+            stdErr + other.stdErr,
+            exitCode + other.exitCode)
+
+    private fun ProcessOutput.check() {
+        val exitCodeMismatch = exitCode != expectedExitStatus
+        if (exitCodeMismatch) {
+            val message = "Expected exit status: $expectedExitStatus, actual: $exitCode"
+            check(expectedFail) { """
+                    |Test failed. $message
+                    |stdout: $stdOut
+                    |stderr: $stdErr
+                    """.trimMargin()
+            }
+            println("Expected failure. $message")
+        }
+
+        val result = stdOut + stdErr
+        val goldValueMismatch = !outputChecker(result.replace(System.lineSeparator(), "\n"))
+        if (goldValueMismatch) {
+            val message = if (::goldValue.isInitialized)
+                "Expected output: $goldValue, actual output: $result"
+            else
+                "Actual output doesn't match with output checker: $result"
+
+            check(expectedFail) { "Test failed. $message" }
+            println("Expected failure. $message")
+        }
+
+        check(!exitCodeMismatch && !goldValueMismatch && !expectedFail) { "Unexpected pass" }
+    }
+}
+
+/**
+ * Executes a standalone tests provided with either @param executable or by the tasks @param name.
+ * The executable itself should be built by the konan plugin.
+ */
+open class KonanStandaloneTest : KonanLocalTest() {
+    init {
+        val target = project.testTarget
+        outputDirectory = "${project.testOutputLocal}/$name"
+        executable = "$outputDirectory/${target.name}/$name.${target.family.exeSuffix}"
+        useFilter = false
+        testLogger = Logger.EMPTY
+    }
+
+    @Optional
+    var enableKonanAssertions = true
+
+    private var _flags: MutableList<String> = mutableListOf()
+    /**
+     * Compiler flags used to build a test.
+     */
+    var flags: MutableList<String>
+        get() {
```

Comment:
It's possible to use `field` here and get rid of `_flags`:
```kotlin
get() {
    if (enableKonanAssertions) field.add("-ea")
    return field
}

set(value) {
    field = value
}
```
Also is't it better to avoid mutation of the backing field? May be make this property just `List<String>` (not mutable) and replace this getter with something like: 

```kotlin
 get() = if (enableKonanAssertions) field + "-ea" else field
```
?